src/html/XHPBaseHTMLHelpers.php

```diff
@@ -0,0 +1,68 @@
+<?hh // strict
+
+/*
+ *  Copyright (c) 2015, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+trait XHPBaseHTMLHelpers implements HasXHPBaseHTMLHelpers {
+  require extends :x:composable-element;
+
+  /*
+   * Appends a string to the "class" attribute (space separated).
+   */
+  public function addClass(string $class): this {
+    try {
+      $current_class = /* UNSAFE_EXPR */ $this->:class;
+      return $this->setAttribute('class', trim($current_class.' '.$class));
```

Comment:
There should be something on the class to denote that a set of attributes should be appended instead of replaced. For normal HTML that's just class, but there can be others. For instance, Javelin needs `data-sigil`. What about putting a function in this trait that is like

```
protected function getAttributesThatAppendValuesOnTransfer(): ImmSet<string> {
  return ImmSet {'string'};
}
```

And you can override it individual classes or create your own trait that extends this to add your special-needs attributes.